omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -109,17 +127,63 @@ public Object create(SimpleObject post, RequestContext context) throws ResponseE
 				case UNLOAD:
 					unloadModules(modules, servletContext);
 					break;
+				case INSTALL:
+					Module module = installModule(modules, installUri, servletContext);
+					List<Module> listModule = new ArrayList<Module>();
+					listModule.add(0, module);
+					action.setModules(listModule);
+					break;
 			}
 		}
-		
+
 		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
 	}
-	
+
+	private Module installModule(Collection<Module> modules, String installUri, ServletContext servletContext) {
+		List<Module> moduleList = new ArrayList<Module>(modules);
+		Module existingModule = moduleList.get(0);
+		Module tempModule = null;
+		File moduleFile = null;
+
+		try {
+			//Module existingModule = moduleFactoryWrapper.getModuleById(module.getModuleId());
```

Comment:
For the above, did you see this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotcommentoutcode